modules/utils/src/timing/stop_watch.cpp

```diff
@@ -0,0 +1,74 @@
+//=================================================================================================
+// PIPPO © Copyright, 2015-2023. All Rights Reserved.
+//
+// This file is part of the RoboCool project
+//=================================================================================================
+
+#include "hephaestus/utils/timing/stop_watch.h"
+
+#include <chrono>
+#include <cstddef>
+#include <tuple>
+
+namespace heph::utils::timing {
+
+StopWatch::StopWatch() {
+  reset();
+}
+
+void StopWatch::start() {
+  if (is_running_) {
+    return;
+  }
+  lap_start_time_ = ClockT::now();
+  if (initial_start_time_.time_since_epoch().count() == 0) {
+    initial_start_time_ = lap_start_time_;
+  }
+  is_running_ = true;
+}
+
+auto StopWatch::lapse() const -> ClockT::duration {
+  if (!is_running_) {
+    return {};
+  }
```

Comment:
This should probably throw. Or you return an optional, but that seems unnecessary.